src/datadog/curl.cpp

```diff
@@ -368,7 +369,7 @@ Expected<void> CurlImpl::post(const HTTPClient::URL &url,
   throw_on_error(curl_.easy_setopt_headerdata(handle.get(), request.get()));
   throw_on_error(curl_.easy_setopt_writefunction(handle.get(), &on_read_body));
   throw_on_error(curl_.easy_setopt_writedata(handle.get(), request.get()));
-  throw_on_error(curl_.easy_setopt_timeout(handle.get(), k_default_timeout_ms));
+  throw_on_error(curl_.easy_setopt_timeout(handle.get(), timeout.count()));
```

Comment:
Use `std::chrono::duration_cast<std::chrono::milliseconds>()` instead because the timeout can be in another unit